blueutil.m

```diff
@@ -51,6 +53,119 @@ int assert_reg(int errcode, const regex_t *restrict preg, char *reason) {
 
 void _NSSetLogCStringFunction(void (*)(const char *, unsigned, BOOL));
 
+// Mock IOBluetoothDevice for subprocess method
+@interface MockBluetoothDevice : NSObject
+@property (nonatomic, strong, readonly) NSString *address;
+@property (nonatomic, strong, readonly) NSString *name;
+@property (nonatomic, assign, readonly) BOOL paired;
+@property (nonatomic, assign, readonly) BOOL connected;
+@property (nonatomic, assign, readonly) BOOL favorite;
+@property (nonatomic, strong, readonly) NSDate *recentAccessDate;
+@property (nonatomic, assign, readonly) int rssi;
+@property (nonatomic, strong, readonly) IOBluetoothDevice *realDevice;
+- (instancetype)initWithAddress:(NSString *)address
+                           name:(NSString *)name
+                         paired:(BOOL)paired
+                      connected:(BOOL)connected
+                       favorite:(BOOL)favorite
+               recentAccessDate:(NSDate *)recentAccessDate
+                           rssi:(int)rssi;
+- (instancetype)initWithAddress:(NSString *)address
+                           name:(NSString *)name
+                      connected:(BOOL)connected
+               recentAccessDate:(NSDate *)recentAccessDate
+                           rssi:(int)rssi;
+- (NSString *)addressString;
+- (BOOL)isConnected;
+- (BOOL)isPaired;
+- (BOOL)isFavorite;
+- (BOOL)isIncoming;
+- (char)RSSI;
+- (char)rawRSSI;
+@end
+
+@implementation MockBluetoothDevice
+- (instancetype)initWithAddress:(NSString *)address
+                           name:(NSString *)name
+                         paired:(BOOL)paired
+                      connected:(BOOL)connected
+                       favorite:(BOOL)favorite
+               recentAccessDate:(NSDate *)recentAccessDate
+                           rssi:(int)rssi {
+  if (self = [super init]) {
+    _address = address;
+    _name = name;
+    _paired = paired;
+    _connected = connected;
+    _favorite = favorite;
+    _recentAccessDate = recentAccessDate;
+    _rssi = rssi;
+    _realDevice = [IOBluetoothDevice deviceWithAddressString:address];
+  }
+  return self;
+}
+
+- (instancetype)initWithAddress:(NSString *)address
+                           name:(NSString *)name
+                      connected:(BOOL)connected
+               recentAccessDate:(NSDate *)recentAccessDate
```

Comment:
This argument is not really used, so can be removed, and `NULL` be sent to next method

Reply:
I've removed it and set to `NULL` internally. This had the effect of not showing any time stamps in the command output. Would you consider removing it completely?

Reply:
You mean entirely from output - probably makes sense given that it is not working, but can be done later